src/unix/index_unix.ml

```diff
@@ -175,9 +175,6 @@ module IO : Index.IO = struct
     else (
       Raw.unsafe_write t.raw ~off:t.flushed buf;
       Raw.Offset.set t.raw offset;
-
-      (* concurrent append might happen so here t.offset might differ
```

Comment:
It's no longer clear to me from reading the code why this is a failure case.

Reply:
Actually, concurrent append cannot happen anymore due to the lock file, the comment was outdated.
The check still stands: after writing the file, if the previously flushed data + the data we just flushed is not the same size as the content written to the disk (header + data), then something is wrong.

Reply:
I see; I didn't understand the purpose of the check then. What sort of runtime behaviour might cause this check to fail? Should it be an assert instead?

Reply:
I can't think of a scenario where this fails (but maybe there is?), and it could indeed be an assert. If I remember well, we had issues with this when the lock file wasn't there yet, and we needed to print for debugging.